### PR TITLE
Handle different error message fields for Auth

### PR DESCRIPTION
**Pull request by djbarnwal**
Auth0 is not consistent with it's error fields. 

For example adding a weak password on SignUp gives the below error as `description` is an object
Screenshot 2023-06-02 at 10 23 19 PM

This PR tries for a best effort solution looking for a string error message.

### CODE REVIEW

File: web-auth/src/components/Auth.svelte [hunk 1]
```diff
@@ -109,8 +109,19 @@
...
             password: password,
           },
-          (err) => {
-            if (err) displayError({ message: err?.description });
+          // explicitly typing as any to avoid missing property TS/svelte-check error
+          (err: any) => {
+            // Auth0 is not consistent in the naming of the error description field
+            const errorText =
+              typeof err?.description === "string"
+                ? err?.description
```
Comment (ericpgreen2) [inline web-auth/src/components/Auth.svelte:117]: Can't we remove the `?`s after we've checked for the existence of each property

Comment (ericpgreen2) [reply web-auth/src/components/Auth.svelte:117]: e.g. `err?.description` --> `err.description`

**Review by ericpgreen2 — APPROVED** [review]
Looks reasonable – approving.

However, I'm seeing that the `displayError(err: any)` function could be made simpler. It takes an untyped `err` object, yet expects the object to have a `message` property. It looks like it could be `displayError(message: string)`. Feel free to make this edit or not.

**Comment by djbarnwal** [comment]
> Looks reasonable – approving.
> 
> However, I'm seeing that the `displayError(err: any)` function could be made simpler. It takes an untyped `err` object, yet expects the object to have a `message` property. It looks like it could be `displayError(message: string)`. Feel free to make this edit or not.

Have made this affordance so that in future we can add more customization to how we display the error messages. For ex - via a toast or showing a `errorTitle` and `errorDescription`.

**Comment by ericpgreen2** [comment]
> Have made this affordance so that in future we can add more customization to how we display the error messages. For ex - via a toast or showing a `errorTitle` and `errorDescription`.

Gotcha. In that case, at least typing the `err` object would be helpful.

**Pull request merged**